lib/pkgcloud/openstack/network/client/securityGroups.js

```diff
@@ -0,0 +1,145 @@
+/*
+ * securityGroups.js: Instance methods for working with security groups
+ * for Openstack Networking
+ *
+ * (C) 2015 Rackspace
+ *      Shaunak Kashyap
+ * MIT LICENSE
+ */
+
+var urlJoin = require('url-join');
+
+var securityGroupsResourcePath = '/security-groups';
+
+// Declaring variables for helper functions defined later
+var _convertSecurityGroupToWireFormat;
+
+/**
+ * client.getSecurityGroups
+ *
+ * @description get the list of security groups for an account
+ *
+ * @param {object|Function}   options
+ * @param {Function}          callback
+ */
+exports.getSecurityGroups = function (options, callback) {
+  var self = this;
+
+  if (typeof options === 'function') {
+    callback = options;
+    options = {};
+  }
+
+  var getSecurityGroupOpts = {
+    path: securityGroupsResourcePath
+  };
+
+  this._request(getSecurityGroupOpts, function (err, body) {
+    if (err) {
+      return callback(err);
+    }
+    else if (!body || !body.security_groups || !(body.security_groups instanceof Array)) {
+      return new Error('Malformed API Response');
+    }
+
+    return callback(null, body.security_groups.map(function (securityGroup) {
+      return new self.models.SecurityGroup(self, securityGroup);
+    }));
+  });
+};
+
+/**
+ * client.getSecurityGroup
+ *
+ * @description get the details for a specific securityGroup
+ *
+ * @param {String|object}     securityGroup     the securityGroup or securityGroupId
+ * @param callback
+ */
+exports.getSecurityGroup = function (securityGroup, callback) {
+  var securityGroupId = securityGroup instanceof this.models.SecurityGroup ? securityGroup.id : securityGroup,
+    self = this;
+  self.emit('log::trace', 'Getting details for security group', securityGroupId);
+  this._request({
+    path: urlJoin(securityGroupsResourcePath, securityGroupId),
+    method: 'GET'
+  }, function (err, body) {
+    if (err) {
+      return callback(err);
+    }
+
+    if (!body ||!body.security_group) {
+      return new Error('Malformed API Response');
+    }
+
+    callback(null, new self.models.SecurityGroup(self, body.security_group));
+  });
+};
+
+/**
+ * client.createSecurityGroup
+ *
+ * @description create a new security group
+ *
+ * @param object      securityGroup
+ * @param string      securityGroup.name        Name of security group.
+ * @param {string}    securityGroup.description Description of security group.
+ * @param {string}    securityGroup.tenantId    The ID of the tenant who owns the security group. Only administrative users can specify a tenant ID other than their own.
+ * @param callback
+ */
+exports.createSecurityGroup = function (securityGroup, callback) {
+  var self = this;
+
+  securityGroup = _convertSecurityGroupToWireFormat(securityGroup);
+
+  var createSecurityGroupOpts = {
+    method: 'POST',
+    path: securityGroupsResourcePath,
+    body: { 'security_group' : securityGroup}
```

Comment:
styling: should have a space after property value, i.e. `{ 'security_group': securityGroup }`